### ISSUE TITLE
Issue on requirements for fla 0.1

### ISSUE THREAD

**Issue by JonathannJacobs** [hustvl/PixelHacker#9]
Hi !
The current requirements.txt lists fla==0.1 and transformers==4.40 while the fla==0.1 module requires transformers>=4.45 as stated here :
https://pypi.org/project/flash-linear-attention/0.1/

This prevents me to build it as when i then try with 4.45 more depencies needs to be updated and it leads to an invalid versions combos overall.

**Comment by SongOnWater** [comment]
"fla" is not an real pip package, and even though flash-linear-attention==0.1 requires torch>=2.5

**Comment by JonathannJacobs** [comment]
Exactly, making fla angry at multiple levels in the dependencies requirements as the torch version is listed here at 2.3
It's possible that fla is only used in a limited scope in pixelhacker and that it won't matter, but if that's the case and is still validated like that, there should be a clean instruction for how to deal with the install.

**Comment by aTunass** [comment]
#6 try it

**Comment by EusdenLin** [comment]
> [#6](https://github.com/hustvl/PixelHacker/issues/6) try it

It works for me. Thanks!

**Comment by JonathannJacobs** [comment]
Well i tried it. In my case i'm trying a slightly more complicated build with cog library and more problems arise later but it's related to flash-attn==2.5.8 and not fla anymore. So this looks like a solution to the ticket issue.
Until someone really commit the requirement.txt fix, i'll keep this ticket open for people to find it.

**Comment by Hwijune** [comment]
too hard to install. 

It seems that torch, flash-attn, fla are not compatible.

**Comment by abhinandan-chakraborty** [comment]
After many hours of test, what worked for me is:
1. Make sure you have Cuda Drivers and Cuda Toolkit installed on your system.
2. Make sure that  the NVIDIA CUDA driver library  is installed on your system in the standard location (/usr/lib/x86_64-linux-gnu/), and there's a symbolic link libcuda.so.1 pointing to it. You can do so by: 
```bash
#make the link
sudo ln -s /usr/lib/x86_64-linux-gnu/libcuda.so.1 /usr/lib/x86_64-linux-gnu/libcuda.so
#verify the link was successful
ls -l /usr/lib/x86_64-linux-gnu/libcuda.so
#update 
sudo ldconfig
```
3. Now create your conda environment and clone the repo and download all the weights
4. then first install `pip install torch==2.6.0 torchvision`
5. Update `requirements.txt`
```requirements.txt
accelerate==0.34.0
diffusers==0.30.2
einops==0.7.0
flash-linear-attention==0.1.0
lightning==2.5.1
lpips==0.1.4
numpy==1.24.4
omegaconf==2.3.0
opencv-python==4.8.0.74
protobuf==3.19.0
pytorch-lightning==2.5.1
PyWavelets==1.6.0
PyYAML==6.0.1
safetensors==0.4.3
scikit-learn==1.2.0
scipy==1.12.0
six==1.16.0
sympy==1.13.1
tensorboard==2.9.0
timm==0.9.11
tokenizers==0.20.0
toml==0.10.2
torch==2.6.0
torchinfo==1.8.0
torchmetrics==1.6.0
torchvision
tqdm==4.66.3
transformers==4.45.0
triton==3.2.0
```
6. Now either install and build `flash-attn` (takes time) OR use a prebuilt version:
```bash
#full build
pip install flash-attn==2.5.8


# ---------------------- OR --------------------


#prebuilt version
pip install https://github.com/mjun0812/flash-attention-prebuild-wheels/releases/download/v0.0.8/flash_attn-2.5.9+cu124torch2.6-cp310-cp310-linux_x86_64.whl
#find your version here: https://github.com/mjun0812/flash-attention-prebuild-wheels?tab=readme-ov-file#packages (so we used 2.5.9 + torch 2.6 + cuda 12.4 + x86 arch + python10)
```
7. All done. Run the script to test it out.

In my experience, the result has not been up to my expectation for a very simple image + small mask so see for yourself if it works for your use case.

**Comment by MrVeryyy** [comment]
> After many hours of test, what worked for me is:
> 
> 1. Make sure you have Cuda Drivers and Cuda Toolkit installed on your system.
> 2. Make sure that  the NVIDIA CUDA driver library  is installed on your system in the standard location (/usr/lib/x86_64-linux-gnu/), and there's a symbolic link libcuda.so.1 pointing to it. You can do so by:
> 
> #make the link
> sudo ln -s /usr/lib/x86_64-linux-gnu/libcuda.so.1 /usr/lib/x86_64-linux-gnu/libcuda.so
> #verify the link was successful
> ls -l /usr/lib/x86_64-linux-gnu/libcuda.so
> #update 
> sudo ldconfig
> 3. Now create your conda environment and clone the repo and download all the weights
> 4. then first install `pip install torch==2.6.0 torchvision`
> 5. Update `requirements.txt`
> 
> accelerate==0.34.0
> diffusers==0.30.2
> einops==0.7.0
> flash-linear-attention==0.1.0
> lightning==2.5.1
> lpips==0.1.4
> numpy==1.24.4
> omegaconf==2.3.0
> opencv-python==4.8.0.74
> protobuf==3.19.0
> pytorch-lightning==2.5.1
> PyWavelets==1.6.0
> PyYAML==6.0.1
> safetensors==0.4.3
> scikit-learn==1.2.0
> scipy==1.12.0
> six==1.16.0
> sympy==1.13.1
> tensorboard==2.9.0
> timm==0.9.11
> tokenizers==0.20.0
> toml==0.10.2
> torch==2.6.0
> torchinfo==1.8.0
> torchmetrics==1.6.0
> torchvision
> tqdm==4.66.3
> transformers==4.45.0
> triton==3.2.0
> 6. Now either install and build `flash-attn` (takes time) OR use a prebuilt version:
> 
> #full build
> pip install flash-attn==2.5.8
> 
> 
> # ---------------------- OR --------------------
> 
> 
> #prebuilt version
> pip install https://github.com/mjun0812/flash-attention-prebuild-wheels/releases/download/v0.0.8/flash_attn-2.5.9+cu124torch2.6-cp310-cp310-linux_x86_64.whl
> #find your version here: https://github.com/mjun0812/flash-attention-prebuild-wheels?tab=readme-ov-file#packages (so we used 2.5.9 + torch 2.6 + cuda 12.4 + x86 arch + python10)
> 7. All done. Run the script to test it out.
> 
> In my experience, the result has not been up to my expectation for a very simple image + small mask so see for yourself if it works for your use case.

Thanks ! for the solutions, It works perfect for me.
But just like you said, the result performance is not that good....
